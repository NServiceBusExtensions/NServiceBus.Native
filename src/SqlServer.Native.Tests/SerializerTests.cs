using System.Collections.Generic;
using ObjectApproval;
using Xunit;
using Xunit.Abstractions;

public class SerializerTests :
    XunitLoggingBase
{
    [Fact]
    public void Dictionary()
    {
        var serialized = Serializer.SerializeDictionary(
            new Dictionary<string, string>
            {
                {"key", "value"},
                {@"a\b", @"a\b"},
                {@"a\\b", @"a\\b"},
                {"a\"b", "a\"b"},
                {"a/b", "a/b"},
                {"a//b", "a//b"},
                {@"a\/b", @"a\/b"}
            });
        ObjectApprover.Verify(Serializer.DeSerializeDictionary(serialized));
    }

    [Fact]
    public void List()
    {
        var serialized = Serializer.SerializeList(
            new List<string>
            {
                "value",
                @"a\b",
                @"a\\b",
                "a\"b",
                "a/b",
                "a//b",
                @"a\/b"
            });
        ObjectApprover.Verify(Serializer.DeSerializeList(serialized));
    }

    public SerializerTests(ITestOutputHelper output) :
        base(output)
    {
    }
}