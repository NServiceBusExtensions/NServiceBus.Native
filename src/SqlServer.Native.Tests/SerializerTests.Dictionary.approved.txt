{
  $type: 'Dictionary<string, string>',
  key: 'value',
  a\b: 'a\b',
  a\\b: 'a\\b',
  a"b: 'a"b',
  a/b: 'a/b',
  a//b: 'a//b',
  a\/b: 'a\/b'
}